ts/src/program/event.ts

```diff
@@ -89,18 +87,19 @@ export class EventParser {
   private handleSystemLog(log: string): [string | null, boolean] {
     // System component.
     const logStart = log.split(":")[0];
+    // Did the program finish executing?
+    if (
+      logStart.match(/^Program (.*) success/g) !== null      
```

Comment:
Following your discord comment [here](https://discord.com/channels/739225212658122886/838515957042774038/854753777068539905):

> It appears the problem is that the previous code makes the assumption that a consumed log will be emitted at the end of cpi.
With the system program, this appears not to be the case.

I removed the "consumed" check as the OR condition crashed your test.
Monitoring the "success" log should be enough, as It looks like a "success" log will always be emitted, no matter if the call is a CPI or a trad call.